test/jdk/javax/swing/JFileChooser/CustomApproveButtonTest.java

```diff
@@ -59,41 +60,48 @@ public void run() {
 
     private CustomApproveButtonTest(String lookAndFeel) {
         System.out.println("Testing Look & Feel : " + lookAndFeel);
-        frame = new JFrame("CustomApproveButtonTest");
+
+        setLookAndFeel(lookAndFeel);
 
         try {
-            UIManager.setLookAndFeel(lookAndFeel);
-        } catch (Exception e) {
-            throw new RuntimeException("Failed to set " + lookAndFeel +
-                    " LookAndFeel. Error : " + e);
-        }
+            frame = new JFrame("CustomApproveButtonTest");
 
-        JFileChooser fileChooser = new JFileChooser();
-        fileChooser.setDialogType(JFileChooser.CUSTOM_DIALOG);
-        frame.add(fileChooser);
-        frame.setVisible(true);
-        frame.pack();
+            JFileChooser fileChooser = new JFileChooser();
+            fileChooser.setDialogType(JFileChooser.CUSTOM_DIALOG);
+            frame.add(fileChooser);
+            frame.pack();
+            frame.setVisible(true);
 
-        JButton customApproveButton = fileChooser.getUI().getDefaultButton(fileChooser);
 
-        if (customApproveButton == null) {
-            fail("Cannot find Custom Approve Button in FileChooser!");
-            return;
-        }
-        if (customApproveButton.getText() == null) {
-            fail("Custom Approve Button Text is null in FileChooser!");
-            return;
-        }
+            JButton customApproveButton = fileChooser.getUI().getDefaultButton(fileChooser);
 
-        if (frame != null) {
-            frame.dispose();
+            if (customApproveButton == null) {
+                fail("Cannot find Custom Approve Button in FileChooser!");
+                return;
+            }
+            if (customApproveButton.getText() == null) {
+                fail("Custom Approve Button Text is null in FileChooser!");
```

Comment:
```suggestion
                fail("Approve button text is null in FileChooser!");
```

Reply:
Updated.

Reply:
I wonder why do you like title-style capitalisation in exception messages.

“Approve Button Text” is appropriate if it's a title of something. Since it's a message, “Approve button text” should be used instead.

Reply:
Updated.